boilerplate/openshift/golang-osd-operator/project.mk

```
@@ -16,5 +16,14 @@ EnableOLMSkipRange?=$(shell sed -n 's/.*EnableOLMSkipRange .*"\([^"]*\)".*/\1/p'
 VERSION_MAJOR?=0
 VERSION_MINOR?=1
 
+ifdef RELEASE_BRANCHED_BUILDS
+    BRANCH_NAME := $(shell git rev-parse --abbrev-ref HEAD)
+    BRANCH_MATCHES := $(shell echo $(BRANCH_NAME) | grep -E '^release-4\.[0-9]{2}$$' >/dev/null && echo true || echo false)
+    ifeq ($(BRANCH_MATCHES),true)
+        VERSION_MAJOR := $(shell echo $(BRANCH_NAME) | sed -E 's/release-([0-9]+)\.([0-9]+)/\1/')
+        VERSION_MINOR := $(shell echo $(BRANCH_NAME) | sed -E 's/release-([0-9]+)\.([0-9]+)/\2/')
+    endif
+endif
```

nit: I would have made these regexes match more closely. But meh.

Also, `grep -q` obviates `>/dev/null`.

But also, using more `make`-native funcs could make this a bit less reliant on what's installed on the running platform. For instance, `$(SED)` is [computed](https://github.com/openshift/boilerplate/blob/e6441d727c4539a549d490c841c6806f3b05cbce/boilerplate/_lib/common.sh#L167-L175) in this project for cross-platform compatibility.

Here's a version that I think does the same thing with only two subshells and no `sed`:
```suggestion
ifdef RELEASE_BRANCHED_BUILDS
    BRANCH_NAME := $(shell git rev-parse --abbrev-ref HEAD | grep -E '^release-[0-9]+\.[0-9]+$$')
    ifneq ($(BRANCH_NAME),)
        SEMVER := $(subst release-,,$(subst ., ,$(BRANCH_NAME)))
        VERSION_MAJOR := $(firstword $(SEMVER))
        VERSION_MINOR := $(lastword $(SEMVER))
    endif
endif
```

---

updated! Much cleaner, thank you!